sdk/burdock/query/sql/private/query.py

```diff
@@ -40,29 +37,31 @@ def execute(self, query_string):
         exact_values = self._execute_exact(query_string)
         return self._apply_noise(*exact_values)
 
-    def _apply_noise(self, subquery, query, syms, types, sens, srs):
+    def _apply_noise(self, subquery, query, syms, types, sens, srs, pct=0.95):
         # if user has selected keycount for outer query, use that instead
         kcc = [kc for kc in subquery.keycount_symbols() if kc[0] != "keycount"]
         if len(kcc) > 0:
             srs["keycount"] = srs[kcc[0][0].lower()]
-        srs = srs.filter("keycount", ">", self.tau ** 2)
+        srs = srs.filter("keycount", ">", self.tau ** 2, srs.bounds)
 
         # add noise to all columns that need noise
         for nsym in subquery.numeric_symbols():
             name, sym = nsym
             name = name.lower()
             sens = sym.sensitivity()
+            mechanism = Laplace(self.epsilon, sens, self.tau)
+            srs.bounds[name] = mechanism.bounds(pct)
```

Comment:
Can collapse out sets and type check

Reply:
Collapsed type checks. Retained check of COUNT sensitivity as it does clamping of COUNT less than 0 to 0 and tau filtering.